rasa/core/actions/loops.py

```diff
@@ -92,3 +90,17 @@ async def deactivate(
     ) -> List[Event]:
         # can be overwritten
         return []
+
+    async def _activate_loop(
+        self,
+        output_channel: "OutputChannel",
+        nlg: "NaturalLanguageGenerator",
+        tracker: "DialogueStateTracker",
+        domain: "Domain",
+        events: List[Event],
+    ) -> List[Event]:
+        if not await self.is_activated(output_channel, nlg, tracker, domain):
```

Comment:
can this line actually stay in `def run()`?

Reply:
Yup, of course, less duplication 😄